### DISCUSSION TITLE
About seperating kernel time when running concurent kernel  workload

### DISCUSSION THREAD

**Discussion by InnoBogus** [sarchlab/mgpusim#120] [Q&A]
I want to get benchmarks' kernel time seperately when running concurent workloads as they always overlap, and i found there is no attribute like ''PID" in the task traced by the tracer,  so where should I start to handle this?

**Comment by syifan** [comment]
This is not easy. We need some fundamental changes to the tracing API to support it. This is on our radar and will likely be supported in v4. 

For now, I recommend you check the driver code and find the point where kernels start and end. Then, you can add prints with the timestamp. With that, you can know the start and end times of each kernel individually. 

Also, you can use `--trace-vis` to collect execution traces and use Daisen to examine the kernel execution.
